doxygen/dox/TechnicalNotes.dox

```diff
@@ -32,7 +32,1026 @@
+purpose (raw data in one file, object headers in another, global heap in a third, etc.)
+in order to improve I/O speeds.
+
+In fact, the number of storage variations is probably larger than the number of methods
+that the HDF5 team is capable of implementing and supporting. Therefore, a Virtual File
+Layer API is being implemented which will allow application teams or departments to design
+and implement their own mapping between the HDF5 format address space and storage, with each
+mapping being a separate file driver (possibly written in terms of other file drivers). The
+HDF5 team will provide a small set of useful file drivers which will also serve as examples
+for those who which to write their own:
+<table>
+<tr>
+<td>#H5FD_SEC2</td><td>This is the default driver which uses Posix file-system functions
+like read and write to perform I/O to a single file. All I/O requests are unbuffered
+although the driver does optimize file seeking operations to some extent.
+</td>
+</tr>
+<tr>
+<td>#H5FD_STDIO</td><td>This driver uses functions from 'stdio.h' to perform buffered I/O to a single file.
+</td>
+</tr>
+<tr>
+<td>#H5FD_CORE</td><td>This driver performs I/O directly to memory and can be
+used to create small temporary files that never exist on permanent storage. This
+type of storage is generally very fast since the I/O consists only of memory-to-memory copy operations.
+</td>
+</tr>
+<tr>
+<td>#H5FD_MPIO</td><td>This is the driver of choice for accessing files in parallel
+using MPI and MPI-IO. It is only predefined if the library is compiled with parallel I/O support.
+</td>
+</tr>
+<tr>
+<td>#H5FD_FAMILY</td><td>Large format address spaces are partitioned into more
+manageable pieces and sent to separate storage locations using an underlying driver
+of the user's choice. \ref H5TOOL_RT_UG can be used to change the sizes of the family
+members when stored as files or to convert a family of files to a single file or vice versa.
+</td>
+</tr>
+</table>
+
+\section sec_vfl_use Using a File Driver
+Most application writers will use a driver defined by the HDF5 library or contributed by another
+programming team. This chapter describes how existing drivers are used.
+
+\subsection subsec_vfl_use_hdr Driver Header Files
+Each file driver is defined in its own public header file which should be included by any
+application which plans to use that driver. The predefined drivers are in header files whose
+names begin with 'H5FD' followed by the driver name and '.h'. The 'hdf5.h' header file includes
+all the predefined driver header files.
+
+Once the appropriate header file is included a symbol of the form 'H5FD_' followed by the
+upper-case driver name will be the driver identification number.(The driver name is by convention
+and might not apply to drivers which are not distributed with HDF5.) However, the value may
+change if the library is closed (e.g., by calling #H5close) and the symbol is referenced again.
+
```

Comment:
There are spacing issues in this file around sentences in ()s inserted into other sentences, some of which have grammar issues.  The file should be reviewed and revised in a future PR.